cmd/syncthing/gui.go

```diff
@@ -789,16 +829,19 @@ func (s *apiService) postSystemConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if to.GUI.Password != s.cfg.GUI().Password {
-		if to.GUI.Password != "" {
-			hash, err := bcrypt.GenerateFromPassword([]byte(to.GUI.Password), 0)
-			if err != nil {
-				l.Warnln("bcrypting password:", err)
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
-			}
+	for i, guiCfg := range s.cfg.GUIs() {
+		toGuiCfg := &to.GUIs()[i]
```

Comment:
What happens if we add or remove a config... This might end up out of index.
I think you need some reasonable way to compare the two types of structs (or have some reliable map key) to do lookups.

Reply:
Do we really wish to support addition/removal of gui configuration in a dynamic way?

Reply:
Why not?

Reply:
We support changing most of the settings online as it stands now.

Reply:
Fine. So, this implies to implement some sort of gui config lookup. I propose to use the `Address` field as key for the lookup. Does it sound good for you?

Reply:
Makes sense as these have to be unique.